src/hooks/EthBalance.tsx

```diff
@@ -1,26 +1,19 @@
 import { readProvider } from 'constants/readProvider'
-import { BigNumber } from 'ethers'
-import { useState } from 'react'
-
-import { usePoller } from './Poller'
+import { useQuery } from 'react-query'
 
 export function useEthBalance(address: string | undefined) {
```

Comment:
@jnoh 

1. I tend to prefer it wrapped in its own hook, that way you can't get the key wrong if hand-coded a `useQuery` in multiple places. To move quickly, I've certainly _not_ done that and haven't had any problems, but it's typically with simpler REST endpoints.
2. `useXyzQuery` is a good idea. I don't have a preference, since a simple inspection of the hook or the return type can show you what's going on. If you feel strongly they should end in `Query`, I can make that change.
3. I don't have a preference here either, since TypeScript's inference should save you most of the time. Something you end up doing with react-query is passing an additional options parameter, which requires it to be typed (see SubgraphQuery.ts:32), it can look quite verbose to see an options param and a return value typed very similarly. The biggest limitation to a weakly-typed react-query usage is a strongly-typed Error value and in-hook key validation. I don't usually have problems with the lack of Error typing as I tend to wrap any "error" interaction in a function that can reliably get me a strongly-typed Error object (since all `catch` statements now return an `unknown` error anyways).

Reply:
If we want to truly wrap react-query, I'd suggest we do something like this:
``` 
export function useEthBalance(address: string | undefined): number {
  const { data: balance } =  useQuery(
      ['eth-balance', address],
      async () => {
        if (!address) {
          throw new Error("Can't fetch ETH balance, address is undefined")
        }
        return readProvider.getBalance(address)
      },
      {
        enabled: !!address,
        staleTime: 30000,
        refetchInterval: 30000,
      },
    )
  }

  return balance
}
```

If we're ok with not fully wrapping react-query and returning react-query types, then I think appending Query to the hook name would go along way for readability and understanding what kind of hook it is

--- 

Not super opinionated on either approach, but I do think we should pick one or the other.

Reply:
Also, please consider this comment non-blocking

Reply:
@jnoh I've done this in the past multiple times and it has ended up becoming a pain. The main reason is you lose a few critical elements from returning _only_ `data`, namely: `isLoading`, `refetch`, and `error`. While it's not _always_ useful to use those and you may end up using only `data` most of the time, it's a good idea to at least expose the other properties in the event you _do_ want to show a loading spinner somewhere.

I'll rename these queries to include the `Query` suffix, I think that's reasonable.